core/src/test/java/io/kestra/plugin/core/flow/ForEachItemCaseTest.java

```diff
@@ -303,11 +303,14 @@ public void restartForEachItem() throws Exception {
                 successLatch.countDown();
             }
         });
+        //Sleep before restarting to make sure the failed execution child tasks are persisted.
+        Thread.sleep(1000);
```

Comment:
So you think the issue is that, as executionService is based on the repository, it didn't contain yet the execution on the correct state?

If so, it would be worth adding a more explainable comment ;) 
Another solution would be to await on the repository to return the execution in the good state, this would avoid a Thread.sleep